### PR TITLE
Register rout

### PR DESCRIPTION
**Pull request by kolya182**

### CODE REVIEW

File: server/routes/unregister.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+const express = require('express');
+const router = express.Router();
+const Device = require('../../db/models/device');
+
+router.route('/')
+  .delete((req, res) => {
+    Device.remove( {deviceID: res.body.deviceID} )
+
+    res.send('DELETE route response');
```
Comment (michaelScoff) [inline server/routes/unregister.js]: Can you use JSON format for consistency?

File: server/routes/register.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      token: req.body.token,
+      wallets: req.body.wallets
+    }
+  );
+    
+  Device.findOneAndUpdate({deviceID: req.body.deviceID}, {wallets: req.body.wallets}, (err, doc) => {
+    if (err) throw err;
+
+    if (doc === null) {
+      device.save((err, object) => {
```
Comment (michaelScoff) [inline server/routes/register.js]: `{}` not being used

File: config/custom-environment-variables.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+    "mongo": {
+        "secret": "MONGODB_URI"
```
Comment (michaelScoff) [inline config/custom-environment-variables.json]: is this should be `url` instead?

File: db/models/device.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
+const mongoose = require('mongoose');
+mongoose.Promise = require('bluebird');
```
Comment (michaelScoff) [inline db/models/device.js]: `mongoose.Promise = require('bluebird')` should be specified in `db/index.js` file

File: server/routes/unregister.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+const express = require('express');
+const router = express.Router();
+const Device = require('../../db/models/device');
+
+router.route('/')
+  .delete((req, res) => {
+    Device.remove( {deviceID: res.body.deviceID} )
```
Comment (michaelScoff) [inline server/routes/unregister.js]: You might want to return response after `remove()` executed

File: server/routes/register.js [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+      })
+      .then(() => {
+        res.status(200).json({status: true, message: "Successfully saved to database"}).send();
```
Comment (michaelScoff) [inline server/routes/register.js]: nit: `Successfully saved` and `Failed to save` should be enough, no need to mention database

File: config/custom-environment-variables.json (hunk 3 above)
Comment (kolya182) [reply config/custom-environment-variables.json]: On Heroku correct env = development. Config library will not found development.json in /config and will fallback to custom-environment-variables.json where it's user responsibility to create env var with key MONGODB_URI

File: db/models/device.js (hunk 4 above)
Comment (kolya182) [reply db/models/device.js]: Will cahnge

File: server/routes/register.js (hunk 6 above)
Comment (kolya182) [reply server/routes/register.js]: +

Comment (kolya182) [reply server/routes/register.js]: ++

**Pull request merged**